DevCycle.SDK.Server.Common/API/DevCycleProvider.cs

```diff
@@ -0,0 +1,78 @@
+using Newtonsoft.Json.Linq;
+using OpenFeature;
+using OpenFeature.Constant;
+using OpenFeature.Model;
+
+namespace DevCycle.SDK.Server.Common.API
+{
+    public class DevCycleProvider : FeatureProvider
+    {
+        private DevCycleBaseClient Client { get; }
+
+        public DevCycleProvider(DevCycleBaseClient client)
+        {
+            Client = client;
+        }
+
+        public override Metadata GetMetadata()
+        {
+            return new Metadata(Client.SdkPlatform);
+        }
+
+        private async Task<ResolutionDetails<T>> EvaluateDevCycle<T>(string flagKey, T defaultValue,
+            EvaluationContext context = null)
+        {
+            var user = DevCycleUser.FromEvaluationContext(context);
+            var variable = await Client.Variable(user, flagKey, defaultValue);
+            return variable.GetResolutionDetails();
+        }
+
+        public override async Task<ResolutionDetails<bool>> ResolveBooleanValue(string flagKey, bool defaultValue,
+            EvaluationContext context = null)
+        {
+            return await EvaluateDevCycle(flagKey, defaultValue, context);
+        }
+
+        public override async Task<ResolutionDetails<string>> ResolveStringValue(string flagKey, string defaultValue,
+            EvaluationContext context = null)
+        {
+            return await EvaluateDevCycle(flagKey, defaultValue, context);
+        }
+
+        public override async Task<ResolutionDetails<int>> ResolveIntegerValue(string flagKey, int defaultValue,
+            EvaluationContext context = null)
+        {
+            return await EvaluateDevCycle(flagKey, defaultValue, context);
+        }
+
+        public override async Task<ResolutionDetails<double>> ResolveDoubleValue(string flagKey, double defaultValue,
+            EvaluationContext context = null)
+        {
+            return await EvaluateDevCycle(flagKey, defaultValue, context);
+        }
+
+        public override async Task<ResolutionDetails<Value>> ResolveStructureValue(string flagKey, Value defaultValue,
+            EvaluationContext context = null)
+        {
+            if (!defaultValue.IsStructure)
+                throw new System.Exception("Cannot call ResolveStructureValue with non-structure Value's");
+            var jsonString = JsonSerializer.Serialize(defaultValue,
+                new JsonSerializerOptions() { Converters = { new OpenFeatureValueJsonConverter() } });
+
+            var newtonsoftJObj = Newtonsoft.Json.JsonConvert.DeserializeObject(jsonString);
+            var user = DevCycleUser.FromEvaluationContext(context);
+            var variable = await Client.Variable(user, flagKey, (JObject)newtonsoftJObj);
+            var jsonValue = Newtonsoft.Json.JsonConvert.SerializeObject(variable.Value);
+            var openFeatureValue = JsonSerializer.Deserialize<Value>(jsonValue,
+                new JsonSerializerOptions() { Converters = { new OpenFeatureValueJsonConverter() } });
```

Comment:
This probably could be done better - but given the nightmare of JSON serializing between `JObject` and `Value` I thought that the converter might be the easier method to clean this up.

Reply:
I was definitely confused at first but after reading it a couple times I get where you are coming from in building it

Reply:
are there any code comments you can add to explain why you needed to do this?

Reply:
sure - I can try

Reply:
This is a challenge both in .NET and Java, where we want to avoid hard dependencies on JSON parsers, or even JSON interfaces that aren't part of the standard library :disappointed:

Reply:
Yeah, the wrapping of the Value object really makes that a nightmare.

The goal here was to make it entirely self contained without extra depends, but the helper class was still needed for transiting between the two easily.